# Theoretical questions.
### Как используются ключевые слова try, throw, catch. Можно ли по-разному обработать разные исключения, вылетающие из одного и того же блока try?
- Если понятно, что в программе может появиться исключение, то его *надо* "выкинуть" с помощью слова `throw`.
- После слова `try` в `{}`пишется код, в котором есть потенциальное исключение, потом пишется `catch` и в `()` пишется тот тип исключений, который хочется поймать, если не нужно конкретное, можно написать `std::exeption` и потом в `{}` используя метод `.what()`, можно понять, что это за исключение.
- Можно по разному обработать разные исключения. Для этого после первого `catch` надо сразу написать второй с другим типом исключения и желанной обработкой.
### Что можно и что нельзя делать, если вы поймали исключение? Что означает нетранзакционность блока try? К каким проблемам это может привести? Как эти проблемы решены в Java и Python?
- Исключение можно вывести на экран, записать либо *делегировать*, но его *нельзя игнорировать*
- Блок `try` *не транзакционный* - это значит, что он может прерываться, не закончив важные действия (не очистит память, после её выделения, например). Если сам программист не отследить свою ошибку до запуска кода, потом будет грустно (может потеряться выделенная память). 
- В Python и Java, например, можно написать `finaly`, в блоке которого можно написать всё, что надо выполнить, и не зависимо от исхода программы это *обязательно* выполниться.
